File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int L = 0;
        int R = numbers.size()-1;
        while(L < R){
            if(numbers[L] + numbers[R] == target){
                return vector<int>{L+1, R+1};
            }else if(numbers[L]+numbers[R] > target){
                R--;
            }else if(numbers[L]+ numbers[R]< target){
                L++;
            }
        }
        return vector<int> {};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        L,R = 0, len(numbers)-1

        while L < R:
            x = numbers[L] + numbers[R]
            if x == target:
                return [L+1, R+1]
            elif x > target: 
                R -= 1
            else:
                L += 1
                


